File: AdventOfCode/Day1/countingCalories.js
const fs = require("fs")

const getInput = () => {
    const data = fs.readFileSync("./input.txt").toString()

    return data.split("\n\n")
}

const getTotalElfCalories = (elfCalories) => {
    const totalElfCalories = elfCalories.split("\n").reduce((accr, curr) => accr += Number(curr), 0)
    
    return totalElfCalories
}

data = getInput().map((input) => getTotalElfCalories(input))

maxValue = data[0]
data.forEach((elm) => {
    if(elm > maxValue) {
        maxValue = elm
    }
})

console.log(maxValue)

File: AdventOfCode/Day1/countingCalories.go
package main

import (
	"fmt"
	"io/ioutil"
	"strconv"
	"strings"
	// "io/ioutil"
)

func getInput() []string {
	data, _ := ioutil.ReadFile("./input.txt")

	return strings.Split(string(data), "\n\n")
}

func getTotalElfCalories(elfCalories string) int {
	totalCalories := 0

	splitElfCalories := strings.Split(elfCalories, "\n")

	for _, calories := range(splitElfCalories) {
		intCalories, _ := strconv.Atoi(calories)
		totalCalories += intCalories
	}

	return totalCalories
}

func main() {
	var totalCalories []int

	for _, calories := range getInput() {
		totalCalories = append(totalCalories, getTotalElfCalories(calories))
	}

	maxValue := totalCalories[0]
	for _, calories := range totalCalories {
		if(calories > maxValue) {
			maxValue = calories
		}
	}

	fmt.Printf("The elf with the most calories has %v calories", maxValue)
}